mcos/mcos.py

```diff
@@ -2,32 +2,38 @@
 import pandas as pd
 from typing import List
 
+from mcos.covariance_transformer import CovarianceMatrixDeNoiser
 from mcos.error_estimator import AbstractErrorEstimator
 from mcos.observation_simulator import AbstractObservationSimulator
 from mcos.optimizer import AbstractOptimizer
 
 
-def simulate_optimizations(obs_simulator: AbstractObservationSimulator, n_sims: int, optimizers: List[AbstractOptimizer],
-             error_estimator: AbstractErrorEstimator, de_noise: bool = True) -> pd.DataFrame:
+def simulate_optimizations(
+        obs_simulator: AbstractObservationSimulator,
+        n_sims: int,
+        optimizers: List[AbstractOptimizer],
+        error_estimator: AbstractErrorEstimator,
+        de_noise: bool = True
```

Comment:
replace this line with `covariance_transformers: List[AbstractConvarianceTransformer]`

Reply:
I also changed the block checking `if de_noise` to 
```
        for transformer in covariance_transformers:
            cov_hat = transformer.transform(cov_hat, obs_simulator.n_observations)
```
is that what we're looking for when applying the transformations?